plugins/orchestrator/src/components/next/ExecuteWorkflowPage.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { InfoCard, Progress } from '@backstage/core-components';
+import {
+  useApi,
+  useRouteRef,
+  useRouteRefParams,
+} from '@backstage/core-plugin-api';
+import { JsonValue } from '@backstage/types';
+
+import { Button, Grid, Typography } from '@material-ui/core';
+
+import { WorkflowDataInputSchemaResponse } from '@janus-idp/backstage-plugin-orchestrator-common';
+
+import { orchestratorApiRef } from '../../api';
+import {
+  executeWorkflowRouteRef,
+  workflowInstanceRouteRef,
+} from '../../routes';
+import { BaseOrchestratorPage } from './BaseOrchestratorPage';
+
+export interface ExecuteWorkflowPageProps {
+  initialState?: Record<string, JsonValue>;
+}
+
+export const ExecuteWorkflowPage = (props: ExecuteWorkflowPageProps) => {
+  const orchestratorApi = useApi(orchestratorApiRef);
+  const { workflowId } = useRouteRefParams(executeWorkflowRouteRef);
+  const [loading, setLoading] = useState(false);
+  const [schemaResponse, setSchemaResponse] =
+    useState<WorkflowDataInputSchemaResponse>();
+  // @ts-ignore:
+  const [formState, setFormState] = useState(props.initialState);
+
+  const navigate = useNavigate();
+  const instanceLink = useRouteRef(workflowInstanceRouteRef);
+
+  useEffect(() => {
+    setLoading(true);
+    orchestratorApi
+      .getWorkflowDataInputSchema(workflowId)
+      .then(response => {
+        setSchemaResponse(response);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, [orchestratorApi, workflowId]);
```

Comment:
Can you try using the `useAsync` hook from react-use, you can see an example of how it is being used in the Workflows tab

Reply:
Thanks for your review! Updated!